### PR TITLE
Move over to slf4j

### PR DESCRIPTION
**Pull request by ketan**

### CODE REVIEW

**Comment by ketan** [comment]
Weird, logging into the CLA app, I see that I've already signed the CLA, but the note above says otherwise.

**Comment by arvindsv** [comment]
Did it sort itself out?

**Comment by arvindsv** [comment]
Or, did you do a re-check?

**Comment by ketan** [comment]
> Did it sort itself out?
> Or, did you do a re-check?

I did a recheck, but it took quite a few minutes to get a green check. Not sure what happened.

File: common/src/com/thoughtworks/go/buildsession/BuildSession.java [hunk 1]
```diff
@@ -335,7 +334,7 @@ private void reportException(Exception e) {
             LOG.error(msg, e);
             printlnSafely(msg);
         } catch (Exception reportException) {
-            LOG.error(format("Unable to report error message - %s.", messageOf(e)), reportException);
+            LOG.error("Unable to report error message - %s.", messageOf(e), reportException);
```
Comment (maheshp) [inline common/src/com/thoughtworks/go/buildsession/BuildSession.java]: Can you change this.

File: common/src/com/thoughtworks/go/buildsession/DownloadFileCommandExecutor.java [hunk 2]
```diff
@@ -54,14 +55,12 @@ public boolean execute(BuildCommand command, BuildSession buildSession) {
 
 
         boolean fileExist = artifact.exists();
-        if (LOG.isDebugEnabled()) {
-            LOG.debug(
-                    "Requesting the file [" + artifact.getAbsolutePath() + "], exist? [" + fileExist + "]");
-        }
+        LOG.debug(
+                "Requesting the file [" + artifact.getAbsolutePath() + "], exist? [" + fileExist + "]");
```
Comment (maheshp) [inline common/src/com/thoughtworks/go/buildsession/DownloadFileCommandExecutor.java]: change this

File: common/src/com/thoughtworks/go/domain/FileHandler.java [hunk 3]
```diff
@@ -49,10 +49,8 @@ public FileHandler(File artifact, String srcFile) {
 
     public String url(String remoteHost, String workingUrl) throws IOException {
         boolean fileExist = artifact.exists();
-        if (LOG.isDebugEnabled()) {
-            LOG.debug(
-                    "Requesting the file [" + artifact.getAbsolutePath() + "], exist? [" + fileExist + "]");
-        }
+        LOG.debug(
+                "Requesting the file [" + artifact.getAbsolutePath() + "], exist? [" + fileExist + "]");
```
Comment (maheshp) [inline common/src/com/thoughtworks/go/domain/FileHandler.java]: change

File: server/src/com/thoughtworks/go/server/materials/ScmMaterialUpdater.java [hunk 4]
```diff
@@ -55,11 +55,9 @@ public void insertLatestOrNewModifications(Material material, MaterialInstance m
                 materialChecker.findLatestModification(folder, material, subprocessExecutionContext) :
                 materialService.modificationsSince(material, folder, list.latestRevision(material), subprocessExecutionContext);
         if (newChanges.isEmpty()) {
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug(
-                        String.format("[Material Update] Did not find any new modifications for material '%s' with flyweight '%s' using working directory '%s'", material, material.getFingerprint(),
-                                folder.getAbsolutePath()));
-            }
+            LOGGER.debug(
+                    String.format("[Material Update] Did not find any new modifications for material '%s' with flyweight '%s' using working directory '%s'", material, material.getFingerprint(),
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/materials/ScmMaterialUpdater.java]: change

File: server/src/com/thoughtworks/go/server/scheduling/ScheduleCheckListener.java [hunk 5]
```diff
@@ -37,26 +37,20 @@ public ScheduleCheckListener(ScheduleCheckCompletedTopic checkCompletedTopic,
...
             LOGGER.error(e.getMessage(), e);
             throw new RuntimeException(e);
         } finally {
             checkCompletedTopic.post(new ScheduleCheckCompletedMessage(message.getPipelineName(), message.trackingId()));
 
             schedulingPerformanceLogger.postingMessageAboutScheduleCheckCompletion(message.trackingId(), message.getPipelineName());
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("Finished posting materialCheckCompletedMessage for pipeline " + message.getPipelineName());
-            }
+            LOGGER.debug("Finished posting materialCheckCompletedMessage for pipeline " + message.getPipelineName());
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/scheduling/ScheduleCheckListener.java]: change

File: server/src/com/thoughtworks/go/server/security/GoExceptionTranslationFilter.java [hunk 6]
```diff
@@ -56,9 +56,7 @@ protected void sendStartAuthentication(ServletRequest request, ServletResponse r
         final Logger logger = LoggerFactory.getLogger(GoExceptionTranslationFilter.class);
         SavedRequest savedRequest = new SavedRequest(httpRequest, getPortResolver());
 
-        if (logger.isDebugEnabled()) {
-            logger.debug("Authentication entry point being called; SavedRequest added to Session: " + savedRequest);
-        }
+        logger.debug("Authentication entry point being called; SavedRequest added to Session: " + savedRequest);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/security/GoExceptionTranslationFilter.java]: change

File: server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java [hunk 7]
```diff
@@ -44,9 +44,7 @@ protected void doFilterHttp(HttpServletRequest request, HttpServletResponse resp
         String header = request.getHeader(AUTHORIZATION);//Token token="ACCESS_TOKEN"
 
         if (header != null) {
-            if (logger.isDebugEnabled()) {
-                logger.debug("Oauth authorization header: " + header);
-            }
+            logger.debug("Oauth authorization header: " + header);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java]: change

File: server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java [hunk 8]
```diff
@@ -55,9 +53,7 @@ protected void doFilterHttp(HttpServletRequest request, HttpServletResponse resp
                     Authentication authResult = authenticationManager.authenticate(authenticationToken);
                     SecurityContextHolder.getContext().setAuthentication(authResult);
                 } catch (AuthenticationException e) {
-                    if (logger.isDebugEnabled()) {
-                        logger.debug("Oauth authentication request for token: " + token, e);
-                    }
+                    logger.debug("Oauth authentication request for token: " + token, e);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java]: change

File: server/src/com/thoughtworks/go/server/service/AgentService.java [hunk 9]
```diff
@@ -316,16 +316,12 @@ public Username agentUsername(String uuId, String ipAddress, String hostNameForD
...
-        }
+        LOGGER.debug("Agent is requesting registration {}", agentRuntimeInfo);
         AgentInstance agentInstance = agentInstances.register(agentRuntimeInfo);
         Registration registration = agentInstance.assignCertification();
         if (agentInstance.isRegistered()) {
             agentConfigService.saveOrUpdateAgent(agentInstance, username);
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("New Agent approved " + agentRuntimeInfo);
-            }
+            LOGGER.debug("New Agent approved " + agentRuntimeInfo);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/AgentService.java]: change

File: server/src/com/thoughtworks/go/server/service/ArtifactsService.java [hunk 10]
```diff
@@ -83,17 +83,13 @@ public void initialize() {
     public boolean saveFile(File dest, InputStream stream, boolean shouldUnzip, int attempt) {
         String destPath = dest.getAbsolutePath();
         try {
-            if (LOGGER.isTraceEnabled()) {
-                LOGGER.trace("Saving file [" + destPath + "]");
-            }
+            LOGGER.trace("Saving file [" + destPath + "]");
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/ArtifactsService.java]: change

File: server/src/com/thoughtworks/go/server/service/ArtifactsService.java [hunk 11]
```diff
@@ -83,17 +83,13 @@ public void initialize() {
...
             if (shouldUnzip) {
                 zipUtil.unzip(new ZipInputStream(stream), dest);
             } else {
                 systemService.streamToFile(stream, dest);
             }
-            if (LOGGER.isTraceEnabled()) {
-                LOGGER.trace("File [" + destPath + "] saved.");
-            }
+            LOGGER.trace("File [" + destPath + "] saved.");
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/ArtifactsService.java]: change

File: server/src/com/thoughtworks/go/server/service/ArtifactsService.java [hunk 12]
```diff
@@ -113,13 +109,9 @@ public boolean saveFile(File dest, InputStream stream, boolean shouldUnzip, int
     public boolean saveOrAppendFile(File dest, InputStream stream) {
         String destPath = dest.getAbsolutePath();
         try {
-            if (LOGGER.isTraceEnabled()) {
-                LOGGER.trace("Appending file [" + destPath + "]");
-            }
+            LOGGER.trace("Appending file [" + destPath + "]");
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/ArtifactsService.java]: change

File: server/src/com/thoughtworks/go/server/service/ArtifactsService.java [hunk 13]
```diff
@@ -113,13 +109,9 @@ public boolean saveFile(File dest, InputStream stream, boolean shouldUnzip, int
...
             systemService.streamToFile(stream, dest);
-            if (LOGGER.isTraceEnabled()) {
-                LOGGER.trace("File [" + destPath + "] appended.");
-            }
+            LOGGER.trace("File [" + destPath + "] appended.");
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/ArtifactsService.java]: change

File: server/src/com/thoughtworks/go/server/service/AutoBuild.java [hunk 14]
```diff
@@ -124,16 +124,12 @@ private MaterialRevisions fanInOnWithFallback(MaterialRevisions originalMaterial
     }
 
     private MaterialRevisions fanInOn(MaterialRevisions originalMaterialRevisions, CruiseConfig cruiseConfig, CaseInsensitiveString targetPipelineName) {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("[Revision Resolution] Fan-In Resolution is active for pipeline " + pipelineName);
-        }
+        LOGGER.debug("[Revision Resolution] Fan-In Resolution is active for pipeline " + pipelineName);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/AutoBuild.java]: change

File: server/src/com/thoughtworks/go/server/service/AutoBuild.java [hunk 15]
```diff
@@ -124,16 +124,12 @@ private MaterialRevisions fanInOnWithFallback(MaterialRevisions originalMaterial
...
         return pipelineService.getRevisionsBasedOnDependencies(originalMaterialRevisions, cruiseConfig, targetPipelineName);
     }
 
     private MaterialRevisions fanInOffTriangleDependency(MaterialRevisions originalMaterialRevisions, CruiseConfig cruiseConfig) {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("[Revision Resolution] Triangle Resolution is active for pipeline " + pipelineName);
-        }
+        LOGGER.debug("[Revision Resolution] Triangle Resolution is active for pipeline " + pipelineName);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/AutoBuild.java]: change

File: server/src/com/thoughtworks/go/server/service/ConsoleService.java [hunk 16]
```diff
@@ -123,9 +123,7 @@ public boolean updateConsoleLog(File dest, InputStream in) throws IOException {
             LOGGER.error("Failed to update console log at : [" + dest.getAbsolutePath() + "]", e);
             return false;
         }
-        if (LOGGER.isTraceEnabled()) {
-            LOGGER.trace("Console log [" + dest.getAbsolutePath() + "] saved.");
-        }
+        LOGGER.trace("Console log [" + dest.getAbsolutePath() + "] saved.");
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/ConsoleService.java]: change

File: server/src/com/thoughtworks/go/server/service/StageService.java [hunk 17]
```diff
@@ -377,9 +377,7 @@ private void populateAuthorsAndMingleCards(List<StageFeedEntry> stageEntries, St
                         }
                     }
                 } else {
-                    if (LOGGER.isDebugEnabled()) {
-                        LOGGER.debug("pipeline not found: " + pipelineForRev);
-                    }
+                    LOGGER.debug("pipeline not found: " + pipelineForRev);
```
Comment (maheshp) [inline server/src/com/thoughtworks/go/server/service/StageService.java]: change

File: server/src/com/thoughtworks/studios/shine/cruise/BackgroundStageLoader.java [hunk 18]
```diff
@@ -64,14 +64,10 @@ public void load() {
 
     public void handle(StageFeedEntry feedEntry, final PipelineInstanceLoader pipelineInstanceLoader) {
         StageIdentifier stageIdentifier = feedEntry.getStageIdentifier();
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("handling stage:<" + stageIdentifier + ">...");
-        }
+        LOGGER.debug("handling stage:<" + stageIdentifier + ">...");
```
Comment (maheshp) [inline server/src/com/thoughtworks/studios/shine/cruise/BackgroundStageLoader.java]: change

File: server/src/com/thoughtworks/studios/shine/cruise/BackgroundStageLoader.java [hunk 19]
```diff
@@ -64,14 +64,10 @@ public void load() {
...
         try {
             stageStorage.save(stageResourceImporter.load(stageIdentifier, new InMemoryTempGraphFactory(), transformerRegistry));
-            if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("importing stage:<" + stageIdentifier + "> finished.");
-            }
+            LOGGER.debug("importing stage:<" + stageIdentifier + "> finished.");
```
Comment (maheshp) [inline server/src/com/thoughtworks/studios/shine/cruise/BackgroundStageLoader.java]: change

File: server/src/com/thoughtworks/studios/shine/cruise/stage/details/JobResourceImporter.java [hunk 20]
```diff
@@ -62,39 +62,29 @@ public JobResourceImporter(String artifactBaseDir, TempGraphFactory graphFactory
     }
 
     public Graph importJob(JobInstance job, final String baseUri) throws GoIntegrationException {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Attempting to import job " + job);
-        }
+        LOGGER.debug("Attempting to import job " + job);
```
Comment (maheshp) [inline server/src/com/thoughtworks/studios/shine/cruise/stage/details/JobResourceImporter.java]: change

File: server/src/com/thoughtworks/studios/shine/cruise/stage/details/JobResourceImporter.java [hunk 21]
```diff
@@ -62,39 +62,29 @@ public JobResourceImporter(String artifactBaseDir, TempGraphFactory graphFactory
...
 
         JobXmlViewModel xmlModel = new JobXmlViewModel(job);
         Graph jobGraph = rdfizer.importURIUsingGRDDL(xmlModel, baseUri);
         URIReference jobURI = jobGraph.getURIReference(xmlModel.httpUrl(baseUri));
         importArtifactsForJob(jobURI, jobGraph);
 
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Done building jobs graph with " + jobGraph.size() + " triples");
-        }
+        LOGGER.debug("Done building jobs graph with " + jobGraph.size() + " triples");
```
Comment (maheshp) [inline server/src/com/thoughtworks/studios/shine/cruise/stage/details/JobResourceImporter.java]: fix all string concatenation in this file

File: server/src/com/thoughtworks/studios/shine/cruise/stage/details/StageResourceImporter.java [hunk 22]
```diff
@@ -73,9 +73,7 @@ public void initialize() {
     }
 
     public Graph load(StageIdentifier stageIdentifier, TempGraphFactory tempGraphFactory, final XSLTTransformerRegistry transformerRegistry) throws GoIntegrationException {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Attempting to import stage with url <" + stageIdentifier + "> !");
-        }
+        LOGGER.debug("Attempting to import stage with url <" + stageIdentifier + "> !");
```
Comment (maheshp) [inline server/src/com/thoughtworks/studios/shine/cruise/stage/details/StageResourceImporter.java]: fix all string concatenation in this file

File: util/src/com/thoughtworks/go/util/NestedJarClassLoader.java [hunk 23]
```diff
@@ -73,9 +73,7 @@ public void run() {
     }
 
     private ClassLoader createLoaderForJar(URL jarURL) {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Creating Loader For jar: " + jarURL);
-        }
+        LOGGER.debug("Creating Loader For jar: " + jarURL);
```
Comment (maheshp) [inline util/src/com/thoughtworks/go/util/NestedJarClassLoader.java]: fix all string concatenation in this file

File: server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java (hunk 7 above)
Comment (ketan) [inline server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java]: This is using a log4j logger.

File: server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java (hunk 8 above)
Comment (ketan) [inline server/src/com/thoughtworks/go/server/security/OauthAuthenticationFilter.java]: This is using a log4j logger.

**Comment by ketan** [comment]
Merging after fixing all of @maheshp's comments.

**Pull request merged**